File: divinity_backend/domain/organizations/currency.py
"""
Monedas soportadas para mostrar montos en toda la app (planes, cuotas,
reportes). Códigos ISO 4217 — lista estable, a diferencia del catálogo de
verticales no hace falta una fuente dinámica ni un endpoint propio.
"""

DEFAULT_CURRENCY = 'COP'

CURRENCY_CHOICES = [
    ('COP', 'Peso colombiano (COP)'),
    ('USD', 'Dólar estadounidense (USD)'),
    ('EUR', 'Euro (EUR)'),
    ('MXN', 'Peso mexicano (MXN)'),
    ('ARS', 'Peso argentino (ARS)'),
    ('CLP', 'Peso chileno (CLP)'),
    ('PEN', 'Sol peruano (PEN)'),
    ('BRL', 'Real brasileño (BRL)'),
]

File: divinity_fronted/src/shared/utils/currency.ts
/** Debe reflejar CURRENCY_CHOICES en divinity_backend/domain/organizations/currency.py */
export const CURRENCY_OPTIONS: { code: string; label: string }[] = [
  { code: 'COP', label: 'Peso colombiano (COP)' },
  { code: 'USD', label: 'Dólar estadounidense (USD)' },
  { code: 'EUR', label: 'Euro (EUR)' },
  { code: 'MXN', label: 'Peso mexicano (MXN)' },
  { code: 'ARS', label: 'Peso argentino (ARS)' },
  { code: 'CLP', label: 'Peso chileno (CLP)' },
  { code: 'PEN', label: 'Sol peruano (PEN)' },
  { code: 'BRL', label: 'Real brasileño (BRL)' },
];

export const DEFAULT_CURRENCY = 'COP';

/** Formatea un monto según la moneda de la organización (por defecto COP). */
export const formatMoney = (amount: string | number, currency: string = DEFAULT_CURRENCY): string => {
  const value = typeof amount === 'string' ? Number(amount) : amount;
  if (Number.isNaN(value)) return '—';
  try {
    return new Intl.NumberFormat('es-CO', { style: 'currency', currency }).format(value);
  } catch {
    // Código de moneda inválido/no soportado por Intl — no debería pasar con la lista curada, pero no reventamos la UI.
    return `${currency} ${value.toLocaleString('es-CO')}`;
  }
};
